app/mailboxes/reply_mailbox.rb

```diff
@@ -26,6 +26,8 @@ def find_relative_conversation
   end
 
   def conversation_uuid_from_to_address
+    return check_forwarded_for_email if mail.to.blank?
```

Comment:
rather than placing the logic in this file, I think we could do some clean-up and implement a method in https://github.com/chatwoot/chatwoot/blob/develop/app/presenters/mail_presenter.rb#L89  . 

We could add a method similar to `original_sender`, called `original_reciever` with a comment on how the `to` could be null in specific cases.
https://github.com/chatwoot/chatwoot/blob/c23e2c23d4151a2879b33e5e74c808565246730f/app/presenters/mail_presenter.rb#L89

replace the mail.to with  `decorated_mail_instance.original_reciever` 

A person new to the codebase could totally miss out on the reasons why we implemented this check.  + all the mail-related logic in one place would help us to handle it for all mailboxes without logic duplication.

Reply:
@sojan-official 
I was thinking that but we have this sequence in the reply mailbox,
```rb
before_processing :conversation_uuid_from_to_address,
                    :find_relative_conversation,
                    :verify_decoded_params,
                    :decorate_mail
```
So didn't get much time to check why we have decorated email after `conversation_uuid_from_to_address` and we need `x-forwarded-for` email in the `conversation_uuid_from_to_address`

If we can spend a little more time on this I can figure out and refactor this. I will convert this to draft for now, I will pick this up after the automation rule changes.

Reply:
@tejaswinichile  sounds good 👍 . I had a quick look and seems like we could do the following

 - We could decouple the account related logic in MailPresenter
 - decorate the mail at the start of the sequence using mail presenter 
 - replace the `mail.x` calls with `decorated_object.x` calls 
 - and initialize the final mail object combined with the account logic at the end of the sequence.  
 
But it seems like a larger effort. so either we could park this issue till we get to it, or we could have a separate tech debt for the same.  Either sounds good to me.